contracts/FIFOMarket.sol

```diff
@@ -301,16 +301,35 @@ contract FIFOMarket is
       address(this),
       batchedIds
     );
+    uint256[] memory holdbackPercentages = _removal.batchGetHoldbackPercentages(
+      batchedIds
+    );
+
     // TODO (Gas Optimization): Declare variables outside of loop
     for (uint256 i = 0; i < batchedIds.length; i++) {
       if (batchedAmounts[i] == batchedBalances[i]) {
         totalNumberActiveRemovals -= 1; // removal used up
       }
       totalActiveSupply -= batchedAmounts[i];
-      uint256 noriFee = (batchedAmounts[i] / 100) * _noriFee;
-      uint256 supplierFee = batchedAmounts[i];
+      uint256 noriFee = (batchedAmounts[i] * _noriFee) / 100;
```

Comment:
should divide after multiplying to avoid precision errors / safe math rounding to 0